### PR TITLE
feat(api): Allow granular max response size config

### PR DESCRIPTION
**Pull request by slowli**
## What ❔

Allows configuring max response size per method for JSON-RPC web servers.

## Why ❔

Makes EN more operator-friendly. Some RPC methods (e.g., `eth_getTransactionReceipt`) may return large responses, while for others, smaller output size is OK.

## Checklist

- [x] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [x] Tests for the changes have been added / updated.
- [x] Documentation comments have been added / updated.
- [x] Code has been formatted via `zk fmt` and `zk lint`.
- [x] Spellcheck has been run via `zk spellcheck`.
- [x] Linkcheck has been run via `zk linkcheck`.

### CODE REVIEW

**Comment by Deniallugo** [comment]
Please add it to file based config as well. 
https://github.com/matter-labs/zksync-era/blob/main/etc/env/file_based/general.yaml

File: core/lib/config/src/configs/api.rs [hunk 1]
```diff
@@ -18,6 +26,106 @@ pub struct ApiConfig {
     pub merkle_tree: MerkleTreeApiConfig,
 }
 
+/// Response size limits for specific RPC methods.
+///
+/// The unit of measurement for contained limits depends on the context. In [`MaxResponseSize`],
+/// limits are measured in bytes, but in configs, limits are specified in MiBs.
+#[derive(Debug, Clone, PartialEq)]
+pub struct MaxResponseSizeOverrides(HashMap<String, NonZeroUsize>);
```
[the remark this answers is not in the document]
Comment (Deniallugo) [reply core/lib/config/src/configs/api.rs:34]: I'd say it's dangerous and ddos prone

File: core/bin/external_node/src/config/mod.rs [hunk 2]
```diff
@@ -158,6 +159,9 @@ pub(crate) struct OptionalENConfig {
     /// Maximum response body size in MiBs. Default is 10 MiB.
     #[serde(default = "OptionalENConfig::default_max_response_body_size_mb")]
     pub max_response_body_size_mb: usize,
+    /// Method-specific overrides in MiBs for the maximum response body size.
+    #[serde(default = "MaxResponseSizeOverrides::empty")]
```
Comment (slowli) [inline core/bin/external_node/src/config/mod.rs:163]: @tomg10 Should we leave the default overrides empty for the EN, or can we come up with a more intelligent default (e.g., `eth_getTransactionReceipt=None`)?

Comment (Deniallugo) [reply core/bin/external_node/src/config/mod.rs:163]: Should we really do it? 
Let's disable non limit responses.

**Review by perekopskiy — APPROVED** [review]

**Review by RomanBrodetski — APPROVED** [review]
Sorry for the delay in review.

The code looks good. As for EN defaults , I don't have a strong opinion - but I don't mind keeping it unbound. Power users will configure it, but average users will only access the EN's API themselves, so this is not such a large concern.

**Comment by slowli** [comment]
Merging as-is (i.e., retaining the default response limit of 10 MB); if necessary, can update in a follow-up PR.

**Pull request merged**